### PR TITLE
[docs] New page for `TimeClock`

### PR DESCRIPTION
**Pull request by flaviendelangle**
## Doc preview

- [Time Clock](https://deploy-preview-7280--material-ui-x.netlify.app/x/react-date-pickers/time-clock/)
- [Time Picker](https://deploy-preview-7280--material-ui-x.netlify.app/x/react-date-pickers/time-picker/)

## What's next ?

- `Date Range Calendar` page
- Cleanup of the `Date Range Picker` and `Date Time Picker` pages

### CODE REVIEW

File: packages/x-date-pickers/src/TimeClock/TimeClock.tsx [hunk 1]
```diff
@@ -163,14 +163,16 @@ export const TimeClock = React.forwardRef(function TimeClock<TDate extends unkno
...
+  const utils = useUtils<TDate>();
 
   const props = useThemeProps({
     props: inProps,
     name: 'MuiTimeClock',
   });
 
   const {
-    ampm = false,
+    ampm = utils.is12HourCycleInCurrentLocale(),
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/TimeClock/TimeClock.tsx:175]: I added a smart fallback to be consistent with the picker behavior

Comment (LukasTy) [reply packages/x-date-pickers/src/TimeClock/TimeClock.tsx:175]: I'd say just go with adding it in this PR.
I reckon that barely anyone was using the standalone Clock component.
But I do still think that we should include a breaking change item in migration guide as well as a changelog entry. The example should probably reference the original component name (was it `Clock`?) as well..? 🤔

**Review by LukasTy — APPROVED** [review]
LGTM, just agreeing with the breaking change and so it would be great to update migration guide at the same time. 👌

File: packages/x-date-pickers/src/TimeClock/TimeClock.tsx (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/TimeClock/TimeClock.tsx:175]: Added

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 2]
```diff
@@ -168,6 +168,18 @@ The `date` prop has been renamed `value` on `MonthCalendar`, `YearCalendar`, `Ti
...
+### Use the 12h/24h format from the locale as the default value of the `ampm` prop on `TimeClock`
+
+The default value of the `ampm` prop changed from `false` to `utils.is12HourCycleInCurrentLocale()`.
+It means that the `TimeClock` component will use a 12h time format for locales where the time is usually displayed with a 12h format.
+
+If you want to keep the previous behavior, you just have to set the `ampm` prop to `false` (components renamed in previous section):
+
+```diff
+- <ClockPicker />
++ <TimeClock ampm />
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:180]: Should it maybe be?
```suggestion
+ <TimeClock ampm={false} />
```
To represent what should be changed to maintain previous behaviour... 🤔

**Review by LukasTy — APPROVED** [review]

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 2 above)
Comment (flaviendelangle) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:180]: :facepalm: Yes, I'm not correctly awake

**Pull request merged**